common/src/file_slice.rs

```diff
@@ -32,6 +34,42 @@ pub trait FileHandle: 'static + Send + Sync + HasLen + fmt::Debug {
     }
 }
 
+#[derive(Debug)]
+pub struct WrapFile(pub File);
+
+#[async_trait]
+impl FileHandle for WrapFile {
+    fn read_bytes(&self, range: Range<usize>) -> io::Result<OwnedBytes> {
+        let mut file = self.0.try_clone()?; // Clone the file to read from it separately
+        let file_len = self.len();
+
+        // Calculate the actual range to read, ensuring it stays within file boundaries
+        let start = range.start;
+        let end = range.end.min(file_len as usize);
+
+        // Ensure the start is before the end of the range
+        if start >= end {
+            return Err(io::Error::new(io::ErrorKind::InvalidInput, "Invalid range"));
+        }
+
+        let mut buffer = vec![0; end - start];
+
+        // Seek to the start position in the file
+        file.seek(io::SeekFrom::Start(start as u64))?;
+
+        // Read the data into the buffer
+        file.read_exact(&mut buffer)?;
+
+        Ok(OwnedBytes::new(buffer))
+    }
+    // todo implement async
+}
+impl HasLen for WrapFile {
+    fn len(&self) -> usize {
+        self.0.metadata().unwrap().len() as usize
```

Comment:
Unwrapping on results of doing I/O seems dangerous? Maybe better to return zero as the length instead?

Reply:
zero may have similar bad side effects. Probably best to preload it or change the trait

Reply:
moved to constructor